server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -239,7 +257,20 @@ private boolean isAllowedFromAddress(CreationMessage creationMessage, List<Strin
                 .orElse(false);
     }
 
-    
+    @VisibleForTesting void validateIsUserOwnerOfMailboxes(CreationMessageEntry entry, MailboxSession session) throws MailboxRightsException {
+        if (containsAnyDelegatedMailbox(entry.getValue().getMailboxIds(), session)) {
+            throw new MailboxRightsException();
+        }
+    }
+
+    private boolean containsAnyDelegatedMailbox(List<String> mailboxIds, MailboxSession session) {
```

Comment:
it seems this function is misnamed as we don't mind if the mailbox is delegated, but only if the user is not the owner of one the mailboxes